duesseldorf.py

```diff
@@ -0,0 +1,163 @@
+               '5': 'mit Schwefel',
+               '6': 'mit Phosphat',
+               '7': 'geschwärzt',
+               '8': 'gewachst',
+               '9': 'mit Süßungsmittel',
+               '10': 'enthält eine Phenylalaninquelle',
+               '11': 'mit Säurungsmittel',
+               '12': 'mit Stabilisatoren',
+               '13': 'mit Phosphorsäure',
+               '14': 'mit Nitritpökelsalz',
+               '15': 'mit Milcheiweiß',
+               '16': 'koffeinhaltig',
+               '17': 'chininhaltig',
+               '18': 'enthält Schwefeldioxid und Sulfite',
+               '19': 'Milch und Milcherzeugnisse / enthält Laktose',
+               '20': 'Glutenhaltiges Getreide sowie daraus hergestellte Erzeugnisse',
+               '21': 'Soja und Sojaerzeugnisse',
+               '22': 'Sellerie und Sellerieerzeugnisse',
+               '23': 'Senf und Senferzeugnisse',
+               '24': 'Sesamsamen und Sesamerzeugnisse',
+               '25': 'Lupine und Lupinenerzeugnisse',
+               '26': 'Erdnüsse und Erdnusserzeugnisse',
+               '27': 'Fisch und Fischerzeugnisse',
+               '28': 'Krebstiere und Krebstiererzeugnisse',
+               '29': 'Weichtiere und Weichtiererzeugnisse',
+               '30': 'Schalenfrüchte und Schalenfruchterzeugnisse',
+               '31': 'Eier und Eierzeugnisse'
+               }
+
+
+def parse_url(url, today=False):
+    canteen = OpenMensaCanteen()
+    document = parse(urlopen(url).read())
+
+    days = ('Montag', 'Dienstag', 'Mittwoch', 'Donnerstag', 'Freitag')
+    for day in days:
+        data = document.find('div', {'data-day': day})
+        date = data.attrs['data-date']
+        parse_day(canteen, date, data)
+
+    return canteen.toXMLFeed()
+
+
+def parse_day(canteen, date, data):
+    # get category names from headings
+    counter_names = data.find_all('h2')
+    # get meals from each category
+    counters = data.find_all('div', class_='counter-table')
+    add_meals_from_counter(canteen, counter_names, counters, date)
+
+
+def add_meals_from_counter(canteen, counter_names, counters, date):
+    i = 0
+    for counter in counters:
+        # get correct category name
+        counter_name = counter_names[i].text.strip()
+        # find all menues for this category
+        menu = counter.find('ul', 'menu')
+        if menu:
+            # get contents of menu
+            items = menu.find_all('li')
+            # parse menu items
+            meal, raw_legend = parse_menu_items(items)
+            # Studentenwerk Düsseldorf does not maintain a easily readable list of additives etc.
+            # need to do our own parsing against a static list
+            legend = parse_legend(raw_legend)
+            # get prices for each role
+            prices = parse_prices(counter.find('ul', 'price'))
+            # add meal to list
+            canteen.addMeal(date, counter_name, meal, notes=legend, prices=prices)
+        i += 1
```

Comment:
This manual counting seems a little brittle. Why don't you iterate over the `counter` divs directly (and merge with the previous function)?